keps/sig-cloud-provider/providers/0023-cloud-provide-azure.md

```diff
@@ -0,0 +1,126 @@
+---
+kep-number: 23
+title: Cloud Provider for Microsoft Azure
+authors:
+  - "@dstrebel"
+owning-sig: sig-cloud-provider
+participating-sigs:
+  - sig-azure
+reviewers:
+  - "@dstrebel"
+  - "@justaugustus"
+  - "@khenidak"
+  - "@feiskyer"
+approvers:
+  - "@feiskyer"
+  - "@khenidak"
+  - "@hogepodge"
+  - "@jagosan"
+editor: TBD
+creation-date: 2019-01-29
+last-updated: 2019-01-29
+status: provisional
+
+---
+
+# Cloud Provider for Microsoft Azure
+
+This is a KEP for adding ```Cloud Provider for Azure``` into the Kubernetes ecosystem.
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Requirements](#requirements)
+* [Proposal](#proposal)
+
+## Summary
+
+Azure provides the Cloud Provider interface implementation as an out-of-tree cloud-controller-manager. It allows Kubernetes clusters to leverage the infrastructure services of Microsoft Azure.
+It is original open sourced project is 
+
+## Motivation
+
+### Goals
+
+The Cloud Provider for Microsoft Azure implements interoperability between Kubernetes clusters and Microsoft Azure. This project will be dedicated to:
+- Provide reliable, secure and optimized integration with Microsoft Azure for Kubernetes
+- Help on the improvement for decoupling cloud provider specifics from Kubernetes core implementation.
+- Providing an open community and environment to for developing Kubernetes on Azure 
+
+
+
+### Non-Goals
+
+//TODO Need More Info
+
+Kubernetes storage support for Azure will be provided by the Azure CSI interface.
+
+E.g.
+
+* [Azure Kubernetes Storage Drivers](https://github.com/Azure/kubernetes-volume-drivers/tree/master/csi)
+
+## Prerequisites
+
+//TODO
```

Comment:
AzureDisk and AzureFile drivers are also depending on the same code path. We may add some words here, that the provider won't be removed until both out-of-tree CSI drivers and cloud provider going GA.